core/src/main/scala/org/apache/spark/resource/ResourceProfile.scala

```diff
@@ -253,6 +265,32 @@ class ResourceProfile(
   }
 }
 
+/**
+ * Resource profile which only contains task resources, used for stage level task schedule when
+ * dynamic allocation is disabled, tasks will be scheduled to executors with default resource
+ * profile based on task resources described by this task resource profile.
+ *
+ * @param taskResources Resource requests for tasks. Mapped from the resource
+ *                      name (e.g., cores, memory, CPU) to its specific request.
+ */
+@Evolving
+@Since("3.4.0")
+class TaskResourceProfile(override val taskResources: Map[String, TaskResourceRequest])
+  extends ResourceProfile(Map.empty, taskResources) {
+
+  /**
+   * Target executor's resource profile id, used for schedule.
+   */
+  override def targetExecutorRpId: Int = ResourceProfile.DEFAULT_RESOURCE_PROFILE_ID
```

Comment:
I think we should override `_id` instead so that `ResourceProfile.getNextProfileId` doesn't increase for task resource profile.

Reply:
I am not sure about this. Since `TaskResourceProfile` is also a special `ResourceProfile` and `_id` is to identify different `ResourceProfile`.